File: nim.c
#include <stdio.h>

struct Mahasiswa {
    char nama[50];
    int nim;
};

struct Mahasiswa dataMahasiswa[3] = {
    {"Jahns Michael", 10},
    {"Indra Nathanael", 11},
    {"Golongan Kami", 12}
};

int main() {
    int SIZE = sizeof(dataMahasiswa)/sizeof(dataMahasiswa[0]);

    int inputNim, found = 0;
    printf("Masukan NIM mahasiswa yang ingin dicari namanya: ");
    scanf("%d", &inputNim);

    for (int i = 0; i < SIZE; i++) {
        if (dataMahasiswa[i].nim == inputNim) {
            printf("Nama mahasiswa dengan NIM %d adalah %s. \n", inputNim, dataMahasiswa[i].nama);
            found = 1;
            break;
        }
    }

    if (found == 0) {
        printf("Pencarian gagal, tidak ada mahasiswa dengan NIM %d atau data tidak lengkap.\n", inputNim);
    }

    return found;
}

File: nim.py
class Mahasiswa:
    def __init__(self, nama, nim):
        self.nama = nama
        self.nim = nim

dataMahasiswa = [
    Mahasiswa("Jahns Michael", 10),
    Mahasiswa("Indra Nathanael", 11),
    Mahasiswa("Golongan Kami", 12),
]

def main():
    found = False
    inputNim = int(input("Masukan NIM mahasiswa yang ingin dicari namanya: "))
    
    for mahasiswa in dataMahasiswa:
        if inputNim == mahasiswa.nim:
            print(f'Nama mahasiswa dengan NIM {mahasiswa.nim} adalah {mahasiswa.nama}.')
            found = True
            break

    if not found:
        print(f'Pencarian gagal, tidak ada mahasiswa dengan NIM {inputNim} atau data tidak lengkap')

if __name__ == "__main__":
    main()
